kyuubi-common/src/main/scala-2.12/scala/util/Using.scala

```diff
@@ -0,0 +1,415 @@
+     *         or the result of the operation if no exceptions were thrown
+     */
+    def apply[A](op: Manager => A): Try[A] = Try { (new Manager).manage(op) }
+
+    private final class Resource[R](resource: R)(implicit releasable: Releasable[R]) {
+      def release(): Unit = releasable.release(resource)
+    }
+  }
+
+  private def preferentiallySuppress(primary: Throwable, secondary: Throwable): Throwable = {
+    def score(t: Throwable): Int = t match {
+      case _: VirtualMachineError => 4
+      case _: LinkageError => 3
+      case _: InterruptedException | _: ThreadDeath => 2
+      case _: ControlThrowable => 0
+      case e if !NonFatal(e) => 1 // in case this method gets out of sync with NonFatal
+      case _ => -1
+    }
+    @inline def suppress(t: Throwable, suppressed: Throwable): Throwable = {
+      t.addSuppressed(suppressed)
+      t
+    }
+
+    if (score(secondary) > score(primary)) suppress(secondary, primary)
+    else suppress(primary, secondary)
+  }
+
+  /** Performs an operation using a resource, and then releases the resource,
+   * even if the operation throws an exception. This method behaves similarly
+   * to Java's try-with-resources.
+   *
+   * $suppressionBehavior
+   *
+   * @param resource the resource
+   * @param body     the operation to perform with the resource
+   * @tparam R the type of the resource
+   * @tparam A the return type of the operation
+   * @return the result of the operation, if neither the operation nor
+   *         releasing the resource throws
+   */
+  def resource[R, A](resource: R)(body: R => A)(implicit releasable: Releasable[R]): A = {
+    if (resource == null) throw new NullPointerException("null resource")
+
+    var toThrow: Throwable = null
+    try {
+      body(resource)
+    } catch {
+      case t: Throwable =>
+        toThrow = t
+        null.asInstanceOf[A] // compiler doesn't know `finally` will throw
+    } finally {
+      if (toThrow eq null) releasable.release(resource)
+      else {
+        try releasable.release(resource)
+        catch { case other: Throwable => toThrow = preferentiallySuppress(toThrow, other) }
+        finally throw toThrow
+      }
+    }
+  }
+
+  /** Performs an operation using two resources, and then releases the resources
+   * in reverse order, even if the operation throws an exception. This method
+   * behaves similarly to Java's try-with-resources.
+   *
+   * $suppressionBehavior
+   *
+   * @param resource1 the first resource
+   * @param resource2 the second resource
+   * @param body      the operation to perform using the resources
+   * @tparam R1 the type of the first resource
+   * @tparam R2 the type of the second resource
+   * @tparam A  the return type of the operation
+   * @return the result of the operation, if neither the operation nor
+   *         releasing the resources throws
+   */
+  def resources[R1: Releasable, R2: Releasable, A](
+                                                    resource1: R1,
```

Comment:
Is scala source so ugly looking?

Reply:
updated codesytle. the original source code at https://raw.githubusercontent.com/scala/scala/v2.13.5/src/library/scala/util/Using.scala